### PR TITLE
feat: add bastion host to allow access to RDS instances

### PR DESCRIPTION
**Pull request by ethanmills**

### CODE REVIEW

**Comment by ethanmills** [comment]
Fixing TFSec and generating a keypair for SSH access

File: terraform/modules/data-share-service/ec2.tf [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    encrypted = true
+  }
+}
+
+resource "aws_key_pair" "rds_bastion_key_pair" {
+  public_key = module.tls_private_key.public_key_pem
+  key_name   = "rds-bastion-key-pair-${var.environment}"
+}
+
+#tfsec:ignore:aws-ec2-no-public-ingress-sgr
```
Comment (oskarwilliams378) [inline terraform/modules/data-share-service/ec2.tf:44]: Is there any sensible limit we can put on this?

File: terraform/modules/data-share-service/rds.tf [hunk 2]
```diff
@@ -48,23 +47,23 @@ resource "aws_rds_cluster_instance" "db_aurora" {
 
 resource "aws_security_group" "rds_postgres_cluster" {
   name_prefix = "${var.environment}-rds-postgres-cluster-"
-  description = "For RDS cluster, inbound access from ECS only"
+  description = "For RDS cluster, inbound access from ECS, bastion host only"
```
Comment (oskarwilliams378) [inline terraform/modules/data-share-service/rds.tf:50]:
```suggestion
  description = "For RDS cluster, inbound access from ECS or bastion host only"
```
Is that what this means?

File: terraform/modules/data-share-service/rds.tf [hunk 3]
```diff
@@ -48,23 +47,23 @@ resource "aws_rds_cluster_instance" "db_aurora" {
...
   vpc_id      = module.vpc.vpc_id
 
   ingress {
     protocol        = "tcp"
     from_port       = 5432
     to_port         = 5432
-    security_groups = [aws_security_group.ecs_tasks.id]
-    description     = "ECS task ingress rule, allow access from ECS tasks only"
+    security_groups = [aws_security_group.ecs_tasks.id, aws_security_group.rds_bastion_host_sg.id]
+    description     = "ECS task ingress rule, allow access from ECS tasks, bastion host only"
```
Comment (oskarwilliams378) [inline terraform/modules/data-share-service/rds.tf:58]: similar comment

**Review by oskarwilliams378 — CHANGES_REQUESTED** [review]

File: terraform/modules/data-share-service/ec2.tf (hunk 1 above)
Comment (ethanmills) [reply terraform/modules/data-share-service/ec2.tf:44]: We could use an IP allow list? I don't know what values we would use other than the SW IP ranges though

Comment (oskarwilliams378) [reply terraform/modules/data-share-service/ec2.tf:44]: for the moment this is good, but could we add a TODO

Comment (ethanmills) [reply terraform/modules/data-share-service/ec2.tf:44]: Created ticket

**Pull request merged**